types/block.go

```diff
@@ -147,18 +150,19 @@ func iterateBlockOpsInTx(op *Op, fetchInputs PrevInputsFetcher, applyOp BlockOpC
 	chainParams *chaincfg.Params) error {
 
 	tx := op.Tx
-	isVote := op.Tree == wire.TxTreeStake && stake.IsSSGen(tx, true)
-	isCoinbase := op.Tree == wire.TxTreeRegular && op.TxIndex == 0
+	isStake := op.Tree == wire.TxTreeStake
+	txType := stake.DetermineTxType(tx, true)
+	isVote := txType == stake.TxTypeSSGen
+	isCoinbase := !isStake && txType == stake.TxTypeRegular && op.TxIndex == 0
 
-	// TODO: Use dcrd's stake.IsTreasuryBase once published instead of this
-	// hack.
-	isTicket := op.Tree == wire.TxTreeStake && !isVote && stake.IsSStx(tx)
-	isTreasuryBase := op.Tree == wire.TxTreeStake && op.TxIndex == 0 && !isVote && !isTicket
+	isTBase := isStake && txType == stake.TxTypeTreasuryBase && op.TxIndex == 0
```

Comment:
I'm fairly certain the tree for the input to a treasury base is _not_ the stake tree, because even though the transaction lives in the stake tree, it references a utxo that doesn't exist, and thus is just the default values.

Reply:
`op.TxIndex` is not the input index but rather the index of the tx in the corresponding tree and `op.Tree` as well is the tree for the entire tx, _not_ for the input.

Reply:
I was referring to the `isStake` portion, because I thought the op was referring to the previous output.  As long as that is for the overall tx, this looks good.

Reply:
Correct, `op` stores what's going to become an individual Rosetta op so it includes information for both the transaction and and individual input or output. `op.Tree` (which is used to determine `isStake`) is the tree of the tx.

In any case, I appreciate the review! :P